### PR TITLE
Add ts-utils-lint

### PR DESCRIPTION
**Pull request by ddknapik**
`ts-utils-lint` wraps all the linting-related libraries, so this would be a single place to take care of updating a dozen+ of packages. TypeScript apps then only need to install this package to have access to the CLI command which runs `eslint` & `prettier` under the hood.

*Please, see readme for more information:* https://github.com/AirHelp/ts-utils/blob/add-ts-utils-lint/ts-utils-lint/README.md

Apart from what's mentioned there, in the upcoming releases we'd like to...
- Polish and document release process (https://github.com/AirHelp/eventboss-js needs it as well).
- Introduce a changelog.
- Set up CI checks.
- Evaluate `tsconfig.json`; current configuration is a seat-of-the-pants[[1]](https://www.merriam-webster.com/dictionary/seat-of-the-pants) solution.

### CODE REVIEW

File: ts-utils-lint/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  ],
+  "parser": "@typescript-eslint/parser",
+  "rules": {
+    "@typescript-eslint/camelcase": [
+      "error",
+      {
+        "properties": "never"
+      }
+    ],
+    "@typescript-eslint/explicit-function-return-type": 0,
```
Comment (kduralek) [inline ts-utils-lint/.eslintrc.json]: It's also nice rule to not to enforce member accessibility on class methods:
`"@typescript-eslint/explicit-member-accessibility": 0`

File: ts-utils-lint/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  preset: 'ts-jest',
+  testEnvironment: 'node',
+  testPathIgnorePatterns: ['node_modules'],
+  testMatch: [ '**/__tests__/**/*.test.ts?(x)']
+};
```
Comment (kduralek) [inline ts-utils-lint/jest.config.js]: `\n` new line

**Comment by ddknapik** [comment]
Thanks @kduralek, I applied your remarks. Regarding file naming convention, this is my only argument against kebab case:

Screenshot 2019-08-06 at 11 20 47

I'm not sure if it's VSCode specific, but I imagine that the _correct_ order would be:

```
ts-utils-lint.ts
ts-utils-lint-copy.ts
```

But _whatever_, let's stick to a single convention 🔥

File: ts-utils-lint/.prettierrc.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "parser": "typescript",
+  "semi": false,
+  "singleQuote": true,
+  "printWidth": 100
```
Comment (managr) [inline ts-utils-lint/.prettierrc.json]: 🎸

File: ts-utils-lint/.prettierrc.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "parser": "typescript",
+  "semi": false,
```
Comment (managr) [inline ts-utils-lint/.prettierrc.json]: hmmm

File: ts-utils-lint/src/__tests__/test-app/.prettierrc.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+  "parser": "typescript",
+  "semi": false,
+  "singleQuote": true
```
Comment (managr) [inline ts-utils-lint/src/__tests__/test-app/.prettierrc.json]: shouldn't we add
```js
 "printWidth": 100
```
here too?

Comment (ddknapik) [reply ts-utils-lint/src/__tests__/test-app/.prettierrc.json]: I believe it doesn't matter what we put in config here (within `test-app`). The point is that executing `tsUtilsLint` on this test project will respect this setup and return appropriate list of errors.

File: ts-utils-lint/.prettierrc.json (hunk 3 above)
Comment (ddknapik) [reply ts-utils-lint/.prettierrc.json]: 🥁

File: ts-utils-lint/.prettierrc.json (hunk 4 above)
Comment (ddknapik) [reply ts-utils-lint/.prettierrc.json]: I believe there's just one reason to use semicolons (I mean, one way to introduce a bug into code by not using semicolons). However, linters protect against it, so let's save those thousands of `;` pressed 💪

**Pull request merged**